test/opt/test_utils.cpp

```diff
@@ -55,4 +55,48 @@ TEST(JoinNonDebugInsts, Cases) {
                  "the only remaining string"}));
 }
 
+namespace {
+typedef struct {
+  const char* orig_str_;
```

Comment:
since this is a plain struct without methods, you can (should) drop the trailing underscore.